go/mysql/flavor.go

```diff
@@ -136,7 +136,7 @@ type flavor interface {
 	baseShowTables() string
 	baseShowTablesWithSizes() string
 
-	supportsCapability(serverVersion string, capability capabilities.FlavorCapability) (bool, error)
+	supportsCapability(capability capabilities.FlavorCapability) (bool, error)
```

Comment:
This now assumes that the serverVersion was specified otherwise. That's fine, I think, but we should probably add some error handling / unit test code covering the case where it's not set.

Reply:
Added unit test as suggested. In actual connections, `serverVersion` is always non-empty; the existing logic for setting a connection's flavor is to return `nil` flavor when there is no known flavor for the given server version (whether server version is empty, or something ridiculous as `"1792.3343.3"`). Just pointing out that on this respect there's nothing new this PR adds.